core/modules/qdisp/ResponseRequester.h

```diff
@@ -101,8 +109,10 @@ class ResponseRequester {
         boost::shared_ptr<CancelFunc> f;
         {
             boost::lock_guard<boost::mutex> lock(_cancelMutex);
-            f.swap(_cancelFunc);
-            _cancelled = true;
+            if(!_cancelled) {
+                f = _cancelFunc;
```

Comment:
I believe Daniel removed that because it the _cancelFunc is called only in the _callCancel(), and inside _CallCancel() we now have a protection through the "_cancelled" flag, which ensures that we only call that function once.  But I agree it'll be safer to set the _cancelFunc to null, just as yet another level of protection. I'll make the change.